modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/BrokerImpl.java

```diff
@@ -625,6 +626,24 @@ public BrokerTransaction newDistributedTransaction() {
         return brokerTransactionFactory.newDistributedTransaction();
     }
 
+    @Override
+    public Set<QueueHandler> restoreDtxPreparedMessages(Xid xid, Collection<Message> messages) throws BrokerException {
+        Set<QueueHandler> queueHandlers = new HashSet<>();
```

Comment:
We might need a read lock here since we are accessing queue handler.

Reply:
Fixed!